glib/src/unichar.rs

```diff
@@ -0,0 +1,239 @@
+    // rustdoc-stripper-ignore-next
+    /// The character could not be decomposed further
+    NoDecomposition,
+    // rustdoc-stripper-ignore-next
+    // A 'singleton' decomposition, which means the character was replaced by another
+    Singleton(char),
+    // rustdoc-stripper-ignore-next
+    /// The first character may decompose further, but the second cannot
+    Pair(char, char),
+}
+
+// rustdoc-stripper-ignore-next
+/// This trait provides access to Unicode character classification and manipulations functions
+/// provided by GLib that do not exist in the standard library
+#[doc(alias = "g_unichar")]
+pub trait Unichar: sealed::Sealed + Copy {
+    #[doc(alias = "g_unichar_type")]
+    #[doc(alias = "unichar_type")]
+    fn unicode_type(self) -> UnicodeType;
+
+    #[doc(alias = "g_unichar_break_type")]
+    #[doc(alias = "unichar_break_type")]
+    fn break_type(self) -> UnicodeBreakType;
+
+    #[doc(alias = "g_unichar_get_script")]
+    #[doc(alias = "unichar_get_script")]
+    fn script(self) -> UnicodeScript;
+
+    #[doc(alias = "g_unichar_combining_class")]
+    #[doc(alias = "unichar_combining_class")]
+    // There's no enum defining named constants for this one as far as I can tell
+    fn combining_class(self) -> i32;
+
+    #[doc(alias = "g_unichar_ismark")]
+    #[doc(alias = "unichar_ismark")]
+    fn is_mark(self) -> bool;
+
+    #[doc(alias = "g_unichar_isgraph")]
+    #[doc(alias = "unichar_isgraph")]
+    fn is_graphical(self) -> bool;
+
+    #[doc(alias = "g_unichar_ispunct")]
+    #[doc(alias = "unichar_ispunct")]
+    fn is_punctuation(self) -> bool;
+
+    #[doc(alias = "g_unichar_istitle")]
+    #[doc(alias = "unichar_istitle")]
+    fn is_titlecase(self) -> bool;
+
+    #[doc(alias = "g_unichar_isdefined")]
+    #[doc(alias = "unichar_isdefined")]
+    fn is_defined(self) -> bool;
+
+    #[doc(alias = "g_unichar_iswide")]
+    #[doc(alias = "unichar_iswide")]
+    fn is_wide(self) -> bool;
+
+    #[doc(alias = "g_unichar_iswide_cjk")]
+    #[doc(alias = "unichar_iswide_cjk")]
+    fn is_wide_cjk(self) -> bool;
+
+    #[doc(alias = "g_unichar_iszerowidth")]
+    #[doc(alias = "unichar_iszerowidth")]
+    fn is_zero_width(self) -> bool;
+
+    #[doc(alias = "g_unichar_totitle")]
+    #[doc(alias = "unichar_totitle")]
+    fn to_titlecase(self) -> char;
+
+    #[doc(alias = "g_unichar_get_mirror_char")]
+    #[doc(alias = "unichar_get_mirror_char")]
+    fn mirror_char(self) -> Option<char>;
+
+    #[doc(alias = "g_unichar_fully_decompose")]
+    #[doc(alias = "unichar_fully_decompose")]
+    fn fully_decompose(self, decomposition_kind: DecompositionKind) -> Vec<char>;
+
+    #[doc(alias = "g_unichar_decompose")]
+    #[doc(alias = "unichar_decompose")]
+    fn decompose(self) -> DecompositionResult;
+
+    #[doc(alias = "g_unichar_compose")]
+    #[doc(alias = "unichar_compose")]
+    fn compose(a: char, b: char) -> Option<char>;
+}
+
+impl Unichar for char {
+    fn unicode_type(self) -> UnicodeType {
```

Comment:
You probably want to add `#[inline]`  here in in other places for these trivial functions to give some more hints to the compiler.